agent_communicator.go

```diff
@@ -173,21 +192,35 @@ func (a *agentCommunicator) sendDataToAgent(suffix string, data interface{}) err
 
 	resp, err := a.client.Do(req)
 
+	if resp == nil {
+		a.l.Debug("Sending data to agent: response nil for URL ", url)
+	}
+
 	if resp != nil {
+		respCode := resp.StatusCode
+		if respCode < 200 || respCode >= 300 {
```

Comment:
Query: Is it better to suggest that there could be an issue based on HTTP status code range ? Also Warn instead of Debug

Reply:
as per our talk, we don't still have a proper handling here, as there is no specific spec to handle http codes returned from the agent here, and this log is purely for troubleshoot purposes (for now atleast)